app/repositories/root_proxies_repository.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+class RootProxiesRepository
+  attr_reader :base_url, :headers
+
+  def initialize(config:)
+    @config = config
+    @base_url = "#{rmapi_url}/rm/rmaccounts/#{config.customer_id}"
+
+    @headers = {
+      'X-Api-Key': config.api_key,
+      'Content-Type': 'application/json',
+      'Accept': 'application/json'
+    }
+  end
+
+  def find!
+    request do
+      status, body = rmapi(:get, '')
+      Result.new(
+        data: to_root_proxy(body[:proxy]),
+        status: status
+      )
+    end
+  end
+
+  def update!(attrs)
+    proxy_types_list = ProxyTypesRepository.new(config: @config).where!
+    proxy_id = attrs[:proxyTypeId]
+    request do
+      root_proxy_validation = Validation::RootProxyParameters.new(proxy_id, proxy_types_list)
+      fail ValidationError, root_proxy_validation unless root_proxy_validation.valid?
+
+      payload = attrs.to_hash.deep_symbolize_keys
+
+      payload[:proxy] = { id: payload.delete(:proxyTypeId) }
+
+      rmapi(:put, '', json: payload)
```

Comment:
Per our discussion regarding packages refactor, we should add some error handling around the PUT such that if it fails for whatever reason, following GET should not be called?

Reply:
You are absolutely correct! on the error handling on PUT's. I was waiting to see the approach that was taking so i can copy that pattern in there.